### PR TITLE
Render radio conditional questions inline rather than below

### PR DESCRIPTION
**Pull request by baarkerlounger**
After this PR:

- When the top level question is a radio button and the conditional question is a text, numeric or date field
   - Then the conditional question is shown inline
- When the top level question is a radio button and the conditional question is a checkbox, select or radio button
  -  Then the conditional question is shown on a new page (not enforced by code)

- When the top level question is a numeric or text question for now the behaviour is unchanged (other household members)

**When the top level question is a radio question**

Before:
image

After:
image

When conditional is not a text box:

image

--------------------------------------------------------------------------------------------------------------------------------------------------------------

**When the top level question is not a radio type the behaviour remain unchanged for now:**
image

Radios with further conditional radio should now be on there own page:
image

image

image

image

### CODE REVIEW

File: app/helpers/question_attribute_helper.rb [hunk 1]
```diff
@@ -20,16 +19,6 @@ def numeric_question_html_attributes(question)
       "data-calculated": question.fields_to_add.to_json,
     }
   end
-
-  def conditional_html_attributes(question)
```
Comment (baarkerlounger) [inline app/helpers/question_attribute_helper.rb]: Can't actually remove this until we also change the Other Household Members conditional questions but after that we can.

**Comment by baarkerlounger** [comment]
@paulrobertlloyd  Few things to confirm here:

1. For the Postcode Known question currently there is no validation for the case `Postcode Known = "No"`, `Postcode entered = "Something"`. Instead we *assume* that if a postcode value has been entered, Postcode Known should have been "Yes" and fix it for them. (This is only really possible via the UI if JS is turned off)

2. Are you happy with conditional check box questions appearing inline as above? If so should there be a max limit on how many answer options it can have or should it just be a new page? (Also still need to make sure input answers are cleared, when invalidated if so)

3. Are you happy with date questions appearing inline as above? 

4. Should there be validations for the care home charges question? I don't see any, nor a ticket for it?

**Comment by paulrobertlloyd** [comment]
1. Let’s validate based on the answer to the radios:
    * If you select ‘Yes’, and have entered a text value for ‘Yes’, validate that text value.
    * If you select ‘No’, but have entered a text value for ‘Yes’, don’t perform validation on that text value.
  
    (The non-JS use case definitely is worth keeping mind)

2. Yes, but I don’t we have or will ever show checkboxes as a conditional field.

3. Yes.

4. Yes. Perhaps check with Mohammed?

**Comment by baarkerlounger** [comment]
@paulrobertlloyd  for (2) we do have that case already - the screenshots above are from the actual form not made up examples

**Comment by paulrobertlloyd** [comment]
2. Sure, but not for long… it’s a separate page in the prototype. 😉

**Comment by baarkerlounger** [comment]
@paulrobertlloyd  for 2 have updated to match.

For 1:

> Let’s validate based on the answer to the radios:
>
>   If you select ‘Yes’, and have entered a text value for ‘Yes’, validate that text value.
>    If you select ‘No’, but have entered a text value for ‘Yes’, don’t perform validation on that text value.

This is how it currently works with the addition that "If you select 'No' it will assume that value is a mistake and clear it for postcodes. 

Will leave it to you to decide if you want to change that in a new ticket?

**Comment by paulrobertlloyd** [comment]
> This is how it currently works with the addition that "If you select 'No' it will assume that value is a mistake and clear it for postcodes.

Perfect.

**Review by dushan-madetech — APPROVED** [review]
Looks good 👍, does the json schema (the one sitting in `config/forms/schema/...`) need to be updated as a result of this change?

**Comment by baarkerlounger** [comment]
@dushan-madetech I don't think so. Nothing that was previously valid isn't valid anymore, we're just documenting that you *shouldn't* do it anymore.

**Pull request merged**